articles/consuming-apis-in-flutter/index.md

```diff
@@ -0,0 +1,250 @@
+Application Programming Interface (API) is a communication portal that allows two or more applications to connect for data sharing. It acts as an intermediary for delivering requests to service providers and returning the responses. The use of APIs has gained prevalence in mobile application development, given the ease of using pre-existing frameworks. Programmers use most APIs to fetch data from web servers and render it to the application's UI components.
+
+This article will demonstrate how to fetch and consume data from a server using a RESTful API in a flutter application. You can download the code for this application form [here](https://github.com/jerimkaura/flutter-book/tree/main/mygithub).
+
+### Prerequisites
+1. A basic understanding of Flutter
+3. Flutter SDK installed on your computer
+2. Code editor, [Android Studio](https://developer.android.com/studio), or [VSCode](https://code.visualstudio.com/download) are most preferred.
+3. An emulator or a mobile device to run the code.
+
+### Table oF contents
+- Setting up the application
+- The HTTP package
+- Creating Data classes from JSON
+- Implementing the providers
+- Consuming the data returned from the API request on a mobile screen
+- Conclusion
+- Further Reading
+
+### Setting up the application
+We will build our application based on the Github API. Therefore, we need to obtain the GitHub `client key` and `secret` to access the API. Check [this link](https://www.knowband.com/blog/user-manual/get-github-client-id-client-secret-api-details/) on a complete guide on getting the `client key` and `secret`.
+
+We need to organize the folders within our flutter project to locate our application's files and components with ease. This practice makes us find bugs easily, unlike writing the entire application on a single file. You can check [this](https://www.section.io/engineering-education/flutter-folder-organization/) article for the preferable folder organization. The final folder organization should appear as below:
+
+```
```

Comment:
How does this folder structure arise? Please explain how to set up the Flutter project from scratch, since this article is targeted towards beginners